src/eve/utils.py

```diff
@@ -289,6 +289,20 @@ def _decorator(base_cls: Type) -> Type:
     return _decorator
 
 
+def abc(original_class: type) -> type:
+    """Mark a class as abstract."""
+
+    class AbstractClass(original_class):  # type: ignore
```

Comment:
I don't agree with this implementation of the decorator: it returns a different class (which is unexpected) and also modifies the inheritance chain. I think there are better approaches here, like for exapmle patching the original class with another `__init__` method which raises an exception. Something like:

```python
def noninstantiable(cls: Type) -> Type:
    def _noninstantiable_init(self, *args, **kwargs) -> None:
        raise TypeError(f"Trying to instantiate `{cls.__name__}` non-instantiable  class.")

    cls.__init__ = _noninstantiable_init
    return cls

```